src/worker/runner/test-runner/index.js

```diff
@@ -21,40 +22,107 @@ module.exports = class TestRunner extends Runner {
         super();
 
         this._test = test.clone();
-        this._test.testplaneCtx = _.cloneDeep(test.testplaneCtx);
+        this._test.testplaneCtx = _.cloneDeep(test.testplaneCtx) || {};
 
         this._file = file;
         this._config = config;
         this._browserAgent = browserAgent;
     }
 
-    async run({ sessionId, sessionCaps, sessionOpts, state, ExecutionThreadCls = ExecutionThread }) {
-        const test = this._test;
-        const testplaneCtx = test.testplaneCtx || {};
+    async run({ sessionId, sessionCaps, sessionOpts, state }) {
```

Comment:
Here I just split this method on three: `prepareToRun`, `runRunnables` and `finishRun`. Code inside them remained the same.